dbms/src/Functions/FunctionsComparison.h

```diff
@@ -684,7 +685,7 @@ class FunctionComparison : public IFunction
             return true;
         };
 
-        if (!callOnBasicTypes<true, false, true, false>(left_number, right_number, call))
+        if (!callOnBasicTypes<true, false, true, true>(left_number, right_number, call))
```

Comment:
If I am not mistaken, it allows to compare date and datetime to decimals.
I think it's ok.

Reply:
It's OK for DateTime cause there're seconds inside. It could be not OK Date cause as I remember SQL requries that whole part of Decimal contains seconds in this case.